pkg/controllers/termination/terminate.go

```diff
@@ -102,34 +95,30 @@ func (t *Terminator) terminate(ctx context.Context, node *v1.Node) error {
 
 // getPods returns a list of pods scheduled to a node based on some filters
 func (t *Terminator) getPods(ctx context.Context, node *v1.Node) ([]*v1.Pod, error) {
-	pods := &v1.PodList{}
-	if err := t.KubeClient.List(ctx, pods, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
+	podList := &v1.PodList{}
+	if err := t.KubeClient.List(ctx, podList, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
 		return nil, fmt.Errorf("listing pods on node, %w", err)
 	}
-	return ptr.PodListToSlice(pods), nil
-}
-
-func (t *Terminator) getEvictablePods(pods []*v1.Pod) []*v1.Pod {
-	evictable := []*v1.Pod{}
-	for _, p := range pods {
+	pods := []*v1.Pod{}
+	for i := range podList.Items {
 		// Ignore if unschedulable is tolerated, since they will reschedule
-		if (v1alpha5.Taints{{Key: v1.TaintNodeUnschedulable, Effect: v1.TaintEffectNoSchedule}}).Tolerates(p) == nil {
+		if (v1alpha5.Taints{{Key: v1.TaintNodeUnschedulable, Effect: v1.TaintEffectNoSchedule}}).Tolerates(&podList.Items[i]) == nil {
 			continue
 		}
 		// Ignore if kubelet is partitioned and pods are beyond graceful termination window
-		if IsStuckTerminating(p) {
+		if IsStuckTerminating(&podList.Items[i]) {
 			continue
 		}
 		// Ignore static mirror pods
-		if pod.IsOwnedByNode(p) {
+		if pod.IsOwnedByNode(&podList.Items[i]) {
 			continue
 		}
-		evictable = append(evictable, p)
+		pods = append(pods, &podList.Items[i])
 	}
-	return evictable
+	return pods, nil
 }
 
-func (t *Terminator) evict(pods []*v1.Pod) {
+func (t *Terminator) evict(ctx context.Context, pods []*v1.Pod) {
```

Comment:
I understand why it's good to have a context here, but if we're not actually using the context that's passed in here and used in `EvictionQueue.Add()`, I don't think there's any difference in code here.

Reply:
Total oversight on my part. I had this wired up for a log line I eventually removed.